[2022-12-26T04:49:24.398+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: sparkify_etl_airflow.Load_songplays_fact_table manual__2022-12-24T11:18:09.055617+00:00 [up_for_retry]>
[2022-12-26T04:49:24.415+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: sparkify_etl_airflow.Load_songplays_fact_table manual__2022-12-24T11:18:09.055617+00:00 [up_for_retry]>
[2022-12-26T04:49:24.415+0000] {taskinstance.py:1283} INFO - 
--------------------------------------------------------------------------------
[2022-12-26T04:49:24.416+0000] {taskinstance.py:1284} INFO - Starting attempt 2 of 4
[2022-12-26T04:49:24.416+0000] {taskinstance.py:1285} INFO - 
--------------------------------------------------------------------------------
[2022-12-26T04:49:24.456+0000] {taskinstance.py:1304} INFO - Executing <Task(LoadFactOperator): Load_songplays_fact_table> on 2022-12-24 11:18:09.055617+00:00
[2022-12-26T04:49:24.462+0000] {standard_task_runner.py:55} INFO - Started process 981 to run task
[2022-12-26T04:49:24.466+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'sparkify_etl_***', 'Load_songplays_fact_table', 'manual__2022-12-24T11:18:09.055617+00:00', '--job-id', '309', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/s3_to_redshift.py', '--cfg-path', '/tmp/tmp50a898p7']
[2022-12-26T04:49:24.467+0000] {standard_task_runner.py:83} INFO - Job 309: Subtask Load_songplays_fact_table
[2022-12-26T04:49:24.566+0000] {task_command.py:389} INFO - Running <TaskInstance: sparkify_etl_airflow.Load_songplays_fact_table manual__2022-12-24T11:18:09.055617+00:00 [running]> on host 8fd7e45d4638
[2022-12-26T04:49:24.806+0000] {taskinstance.py:1513} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=bala
AIRFLOW_CTX_DAG_ID=sparkify_etl_***
AIRFLOW_CTX_TASK_ID=Load_songplays_fact_table
AIRFLOW_CTX_EXECUTION_DATE=2022-12-24T11:18:09.055617+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-12-24T11:18:09.055617+00:00
[2022-12-26T04:49:24.807+0000] {load_fact.py:33} INFO - Loading fact table 'songplays' into Redshift
[2022-12-26T04:49:24.822+0000] {base.py:73} INFO - Using connection ID 'redshift' for task execution.
[2022-12-26T04:49:25.721+0000] {sql.py:364} INFO - Running statement: 
        INSERT INTO songplays
        
        SELECT
                md5(events.sessionid || events.start_time) songplay_id,
                events.start_time, 
                events.userid, 
                events.level, 
                songs.song_id, 
                songs.artist_id, 
                events.sessionid, 
                events.location, 
                events.useragent
                FROM (SELECT TIMESTAMP 'epoch' + ts/1000 * interval '1 second' AS 
start_time, *
            FROM staging_events
            WHERE page='NextSong') events
            LEFT JOIN staging_songs songs
            ON events.song = songs.title
                AND events.artist = songs.artist_name
                AND events.length = songs.duration
    ;
    , parameters: None
[2022-12-26T04:49:25.929+0000] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/airflow/plugins/operators/load_fact.py", line 34, in execute
    redshift.run(formatted_sql)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 338, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 369, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.FeatureNotSupported: cannot set an identity column to a value

[2022-12-26T04:49:26.392+0000] {taskinstance.py:1327} INFO - Marking task as UP_FOR_RETRY. dag_id=sparkify_etl_***, task_id=Load_songplays_fact_table, execution_date=20221224T111809, start_date=20221226T044924, end_date=20221226T044926
[2022-12-26T04:49:26.868+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 309 for task Load_songplays_fact_table (cannot set an identity column to a value
; 981)
[2022-12-26T04:49:26.929+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2022-12-26T04:49:27.013+0000] {taskinstance.py:2582} INFO - 0 downstream tasks scheduled from follow-on schedule check
